server/src/test/java/org/apache/kafka/server/KRaftClusterTest.java

```
@@ -644,6 +674,544 @@ private Admin createAdminClient(KafkaClusterTestKit cluster, boolean usingBootst
+        return changes.keySet().stream().map(resource -> {
+            try {
+                values.get(resource).get();
+                return ApiError.NONE;
+            } catch (ExecutionException e) {
+                return ApiError.fromThrowable(e.getCause());
+            } catch (Throwable t) {
+                return ApiError.fromThrowable(t);
+            }
+        }).collect(Collectors.toList());
+    }
+
+    private Map<ConfigResource, Map<String, String>> validateConfigs(
+        Admin admin,
+        Map<ConfigResource, Map<String, String>> expected,
+        boolean exhaustive
+    ) throws Exception {
+        Map<ConfigResource, Map<String, String>> results = new HashMap<>();
+        TestUtils.retryOnExceptionWithTimeout(60000, () -> {
+            try {
+                var values = admin.describeConfigs(expected.keySet()).values();
+                results.clear();
+                assertEquals(expected.keySet(), values.keySet());
+                for (Map.Entry<ConfigResource, Map<String, String>> entry : expected.entrySet()) {
+                    ConfigResource resource = entry.getKey();
+                    Map<String, String> expectedPairs = entry.getValue();
+                    var config = values.get(resource).get();
+                    Map<String, String> actualMap = new TreeMap<>();
+                    Map<String, String> expectedMap = new TreeMap<>();
+                    config.entries().forEach(configEntry -> {
+                        actualMap.put(configEntry.name(), configEntry.value());
+                        if (!exhaustive) {
+                            expectedMap.put(configEntry.name(), configEntry.value());
+                        }
+                    });
+                    expectedMap.putAll(expectedPairs);
+                    assertEquals(expectedMap, actualMap);
+                    results.put(resource, actualMap);
+                }
+            } catch (Exception t) {
```

Is this for local debugging? Otherwise, we should let the exception propagate instead of catching it